CTFd/challenges.py

```diff
@@ -36,7 +36,7 @@ def chals():
             if view_after_ctf():
                 pass
             else:
-                return redirect('/')
+                return redirect(url_for('/'))
```

Comment:
I don't have experience deploying under gunicorn, so I don't know how that's supposed to be handled.

Based on my limited experience, it appears that you want ``SCRIPT_NAME`` and ``PATH_INFO`` to be managed at the wsgi level (whether that be uwsgi, gunicorn, or tornado) - I'm just not sure how that works for each of the different setups.

I agree that it smells a bit, but until I get a chance to try under gunicorn I'm inclined to go with behavior that is known good with uwsgi (which seems to be a somewhat standard/supported deployment option at the distro level)